Feat/links

freesteph opened the pull request:
Première partie de la réécriture des liens 🎉

adipasquale commented on app/helpers/dsfr_link_helper.rb:
```diff
@@ -1,113 +1,51 @@
-
-    [default_class] + BUTTON_STYLES.values_at(*styles).compact
-  end
+  ICON_SIDES = %w(left right).freeze
```
I'm guessing `position` would be a better name, wdyt?

freesteph replied:
agreed

adipasquale commented on guide/content/helpers/link.slim:
```diff
@@ -0,0 +1,67 @@
+  caption: "Lien au fil du texte",
+  code: link_embed) do
+  markdown:
+    Liens sans attributes particuliers, au fil du texte. Ils n'ont pas
```
attributs

adipasquale commented on guide/layouts/partials/related-info.slim:
```diff
@@ -3,4 +3,4 @@ h2
 ul
   - links.each do |text, link|
-    li== link_to text, link
+    li== link_to text, link, {target: '_blank', rel: 'noopener'}
```
👍

adipasquale replied:
nitpick: curly brackets are optional here

adipasquale reviewed (commented):
👍 well done I think it's quite simple now, both to use and to read the code 

rubocop is not happy with a few things, I think it's confusing `href: #` for a comment start, and other things. 
There might still be something mysterious in its config if you don't have the same behaviour locally.

optionals : 

1. could you add a make command or some doc somewhere on how to use `guard` ? It's been a looong while i  haven't worked with it
2. It would be nice to add some yard doc on the accepted keyword arg options for dsfr_link_to if you're up for it 
3. Icons don't show up in the doc but that's independent from your PR, we'll have to fix it
4.  in slim guide templates `==` does not escape html whereas `=` does. But for html_safe helpers such as dsfr_link_to / link_to it's enough to use `=` (or I'm missing something)

freesteph commented:
> 👍 well done I think it's quite simple now, both to use and to read the code

Thanks!

> 
> rubocop is not happy with a few things, I think it's confusing `href: #` for a comment start, and other things. There might still be something mysterious in its config if you don't have the same behaviour locally.

I've fixed Rubocop and it didn't mention `href`, strange.

By the way I'm trialling a new Git approach where I do a bunch of fixup commits so we don't loose our diff, and then once we're both happy with it I can interactive-rebase and merge.
> 
> optionals :
> 
>     1. could you add a make command or some doc somewhere on how to use `guard` ? It's been a looong while i  haven't worked with it

done

> 
>     2. It would be nice to add some yard doc on the accepted keyword arg options for dsfr_link_to if you're up for it
> 
Yep, sorry forgot about this bit. Done now.

>     3. Icons don't show up in the doc but that's independent from your PR, we'll have to fix it

Right so this is an absolute pain in the neck but basically the DSFR looks up icons at two different paths which makes it incredibly tricky to serve. I am working on a fix but let's save it for another day.
> 
>     4. in slim guide templates `==` does not escape html whereas `=` does. But for html_safe helpers such as dsfr_link_to / link_to it's enough to use `=` (or I'm missing something)

Fixed it thanks.

If you want to have another look, I'm all good.

adipasquale reviewed (approved):
🚀 well done thanks for the quick fixes 🙇
